### PR TITLE
feat: add code samples and tests for overlay creation

### PR DESCRIPTION
**Pull request by irataxy**
Adding in Java samples for https://cloud.google.com/transcoder/docs/how-to/create-overlays

### CODE REVIEW

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    originalOut = System.out;
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
```
Comment (shubha-rajan) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java:102]: duplicated code?
```suggestion
    originalOut = System.out;
    bout = new ByteArrayOutputStream();
    System.setOut(new PrintStream(bout));
```

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    deleteBucket(BUCKET_NAME);
```
Comment (shubha-rajan) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java:104]: Do you need to delete the bucket here if it's deleted in the `tearDown` method?

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  private static void deleteBucket(String bucketName) {
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Bucket bucket = storage.get(bucketName);
+    if (bucket != null) {
+      Page<Blob> blobs = bucket.list();
+
+      for (Blob blob : blobs.iterateAll()) {
+        System.out.println(blob.getName());
```
Comment (shubha-rajan) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java:81]: is it necessary to print the blob names before deleting them?

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import com.google.cloud.video.transcoder.v1beta1.Overlay.AnimationStatic;
+import com.google.cloud.video.transcoder.v1beta1.Overlay.NormalizedCoordinate;
+import com.google.cloud.video.transcoder.v1beta1.TranscoderServiceClient;
+import com.google.cloud.video.transcoder.v1beta1.VideoStream;
+import com.google.protobuf.Duration;
+import java.io.IOException;
+
+public class CreateJobWithStaticOverlay {
+
+  public static void main(String[] args) throws Exception {
```
Comment (shubha-rajan) [inline media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java:41]:
```suggestion
  public static void main(String[] args) throws IOException {
```

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import com.google.cloud.video.transcoder.v1beta1.Overlay.FadeType;
+import com.google.cloud.video.transcoder.v1beta1.Overlay.NormalizedCoordinate;
+import com.google.cloud.video.transcoder.v1beta1.TranscoderServiceClient;
+import com.google.cloud.video.transcoder.v1beta1.VideoStream;
+import com.google.protobuf.Duration;
+import java.io.IOException;
+
+public class CreateJobWithAnimatedOverlay {
+
+  public static void main(String[] args) throws Exception {
```
Comment (shubha-rajan) [inline media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java:42]:
```suggestion
  public static void main(String[] args) throws IOException {
```

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      Overlay.Animation animationFadeOut =
+          Animation.newBuilder()
+              .setAnimationFade(
+                  AnimationFade.newBuilder()
+                      .setFadeType(FadeType.FADE_OUT)
+                      .setXy(NormalizedCoordinate.newBuilder().setX(0.5).setY(0.5).build())
+                      .setStartTimeOffset(Duration.newBuilder().setSeconds(12).build())
+                      .setEndTimeOffset(Duration.newBuilder().setSeconds(15).build())
+                      .build())
+              .build();
```
Comment (shubha-rajan) [inline media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java:96]: This is a really large chunk of setup/boilerplate code. Maybe add newlines between each step and a comment explaining what the step is doing?

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+                      .build())
+              .build();
+      Overlay.Animation animationEnd =
+          Overlay.Animation.newBuilder()
+              .setAnimationEnd(
+                  AnimationEnd.newBuilder()
+                      .setStartTimeOffset(Duration.newBuilder().setSeconds(10).build())
+                      .build())
+              .build();
+      Overlay overlay =
```
Comment (shubha-rajan) [inline media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java:91]: nit: Add a newline before this to separate it from the setup steps before it.

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java [hunk 8]
```diff
@@ -122,7 +122,7 @@ public void test_CreateJobFromAdHoc() throws Exception {
     JOB_ID = arr[arr.length - 1].replace("\n", "");
     bout.reset();
 
-    Thread.sleep(180000);
+    Thread.sleep(300000);
```
Comment (shubha-rajan) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:125]: Is there a reason we're updating the timeout for these tests in this PR? Might be better to update timeouts for other tests in a separate PR.

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    originalOut = System.out;
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
```
Comment (shubha-rajan) [inline media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java:102]:
```suggestion
    originalOut = System.out;
    bout = new ByteArrayOutputStream();
    System.setOut(new PrintStream(bout));

```

**Review by shubha-rajan — CHANGES_REQUESTED** [review]

**Comment by shubha-rajan** [comment]
Hi @irataxy, thanks for the PR. Took a first pass at reviewing

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java (hunk 2 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java:104]: Yes, this is to make sure there is no bucket to begin with (for example, if a test failed partway through).

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java (hunk 1 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobWithAnimatedOverlayTest.java]: Removed duplicate code

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java (hunk 3 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java]: Removed

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java (hunk 4 above)
Comment (irataxy) [reply media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java]: Changed

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java (hunk 5 above)
Comment (irataxy) [reply media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java]: Changed

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java (hunk 6 above)
Comment (irataxy) [reply media/transcoder/src/main/java/com/example/transcoder/CreateJobWithAnimatedOverlay.java]: Done

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java (hunk 9 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobWithStaticOverlayTest.java:101]: Removed duplicate code

File: media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java (hunk 8 above)
Comment (irataxy) [reply media/transcoder/src/test/java/com/example/transcoder/CreateJobFromAdHocTest.java:125]: Not sure why this is showing up. This was already updated in https://github.com/GoogleCloudPlatform/java-docs-samples/commit/8ebe2b9896ccae10205189786b1bf8b905f7fd39

File: media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java (hunk 7 above)
Comment (irataxy) [reply media/transcoder/src/main/java/com/example/transcoder/CreateJobWithStaticOverlay.java:105]: Done

**Review by shubha-rajan — APPROVED** [review]

**Comment by shubha-rajan** [comment]
LGTM, thanks!

**Pull request merged**